### PR TITLE
waf: make implicit fallthroughs fatal

### PR DESCRIPTION
**Pull request by peterbarker**
This seems to be the default on some compilers but not others.

### CODE REVIEW

File: Tools/ardupilotwaf/boards.py [hunk 1]
```diff
@@ -140,11 +140,17 @@ def configure_env(self, cfg, env):
...
                 '-Wno-mismatched-tags',
                 '-Wno-gnu-variable-sized-type-not-at-end',
+                '-Werror=implicit-fallthrough',
             ]
         else:
             env.CFLAGS += [
                 '-Wno-format-contains-nul',
             ]
+            (major, minor, patchlevel) = cfg.env.CC_VERSION
+            if int(major) > 7 or (int(major) == 7 and int(minor) >= 4):
```
Comment (tridge) [inline Tools/ardupilotwaf/boards.py]: how does this work with clang?

File: Tools/ardupilotwaf/boards.py [hunk 2]
```diff
@@ -237,6 +244,10 @@ def configure_env(self, cfg, env):
                 env.CXXFLAGS += [
                     '-Werror=suggest-override',
                 ]
+            if int(major) > 7 or (int(major) == 7 and int(minor) >= 4):
```
Comment (tridge) [inline Tools/ardupilotwaf/boards.py]: can we break out these two into a function?

Comment (tridge) [reply Tools/ardupilotwaf/boards.py]: maybe is_gcc_version_above(7,4)

**Comment by rmackay9** [comment]
This doesn't seem super critical so I'm going to leave it out of Copter-4.0.x

**Pull request merged**